Pod/Native/ParseState.swift

```diff
@@ -91,7 +90,7 @@ class ParseState {
                 
                 // The scenario description must be unique
                 let description = "\(description)_line\(example.lineNumber)"
-                scenarios.append(NativeScenario(description, steps: steps))
+                scenarios.append(NativeScenario(description, steps: steps, index: index + exampleIndex))
```

Comment:
Mhm, well in that case the indices would look like:

```
B: 0
A Alice: 1
A Bob: 2
A Eve: 3
```

because B is before A in the feature file.

The scenario you're describing doesn't happen because after each run of this method we return new scenarios that get aggregated to the scenarios in `NativeFeature`. In your example, it would look roughly like this:

- Feature file starts parsing line by line (inside `NativeFeature`)
- Arrives at first scenario (B) at index 0
- Parses it by passing index (0), and `NativeFeature.scenarios`'s count increases by one 
- Index is now 1
- Arrives at second scenario
- Parses it, passes index (1), and because it's an example, it enumerates each as a scenario (https://github.com/net-a-porter-mobile/XCTest-Gherkin/pull/65/files#diff-25c63a89b3f1147361ff53788c5de3efR77) and adds the passed in index with the index in the loop (https://github.com/net-a-porter-mobile/XCTest-Gherkin/pull/65/files#diff-25c63a89b3f1147361ff53788c5de3efR93).
- The result of parsing the above returns an _array_ of scenarios, so the `NativeFeature.scenarios`'s count now becomes 4 (B + 3 from A). The next scenario will have index 4.

Hopefully this make sense